abci/cmd/abci-cli/abci-cli.go

```diff
@@ -222,6 +224,14 @@ var versionCmd = &cobra.Command{
 	},
 }
 
+var prepareProposalCmd = &cobra.Command{
+	Use:   "prepare_proposal",
+	Short: "prepare proposal",
+	Long:  "prepare proposal",
+	Args:  cobra.ExactArgs(1),
```

Comment:
I see you are iterating over the arguments below, but here you specify "ExactArgs(1)".
Consider changing it to what `FinallizeBlock` has